### ISSUE TITLE
`IntervalMetadata` should allow the same start and end in bounds

### ISSUE THREAD

**Issue by RNAer** [biocore/scikit-bio#1441]
currently, we disallow it. For example,

```
from skbio.metadata import IntervalMetadata
im = IntervalMetadata(3)
im.add([(1, 1)])
```

```
scikit-bio/skbio/metadata/_interval.py in _assert_valid_bound(bound)
    932                 isinstance(end, int)) or start >= end:
    933             raise ValueError('`start` (%r) must be a smaller int '
--> 934                              'than `end` (%r).' % (start, end))
    935     else:
    936         raise TypeError("Each `bound` must be a tuple, not {!r}".format(

ValueError: `start` (1) must be a smaller int than `end` (1).
```

@jairideout , we talked this over the `IntervalMetadata` PR and decided then not to allow it. However, I think we should allow this for annotation of `a site between two indicated adjoining bases. e.g. 123^124`, which is one type of location specified in genbank and other formats. For this example, it can be represented by `[(123, 123)]`.

what do you think?

**Comment by jairideout** [comment]
Sounds good, I think we disallowed b/c there wasn't a clear use-case for a non-position. @ebolyen and I played around with querying those types of intervals and the behavior was kinda odd, though I don't have specific details about that. When you're implementing this, can you make sure these types of intervals are query-able?

**Comment by RNAer** [comment]
sure. I play with it and make sure it queryable.

**Comment by jairideout** [comment]
Thanks!

**Comment by RNAer** [comment]
The current behavior is like this:
```
n [2]: a = IntervalMetadata(9)

In [3]: a.add([(1,1)])
Out[3]: Interval(interval_metadata=<4580885280>, bounds=[(1, 1)], fuzzy=[(False, False)], metadata={})

In [9]: list(a.query([(0, 1)]))
Out[9]: []
In [6]: list(a.query([(1, 2)]))
Out[6]: []
In [7]: list(a.query([(1, 1)]))
Out[7]: []

In [8]: list(a.query([(0, 2)]))
Out[8]: [Interval(interval_metadata=<4580885280>, bounds=[(1, 1)], fuzzy=[(False, False)], metadata={})]
```

It is a little surprising but not totally unexpected, because to determine if the query overlaps with a interval, it uses `self.end > start  and  self.start < end` (where `self` is the interval). Thus, you have to give `(p-1, p+1)` to retrieve the interval of position `p`. 

I can modify the cython code _intersection.pyx a bit to make `a.query([(1, 2)])` return the Interval object. This should make the behavior more expected.

What do you think?

**Comment by jairideout** [comment]
Fixed in #1445.
